app/models/plate.rb

```diff
@@ -373,6 +373,30 @@ def with(subinclude)
     ])
   }
 
+  def self.search_for_plates(study_id: nil, plate_purpose_ids: nil, start_date: nil, end_date: nil)
+    with_study_id(study_id)
+      .with_plate_purpose_ids(plate_purpose_ids)
+      .created_after(start_date)
+      .created_before(end_date)
+      .distinct
+  end
+
+  scope :with_study_id, (proc { |study_id|
```

Comment:
Try swithing to lambda syntax: eg:

->(variable) { block_code }

variable optional. Can drop brackets entirely is its not present.